geosnap/harmonize/harmonize.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+from tobler.area_weighted import *
+from tobler.nlcd import _check_presence_of_crs
+
+def harmonize(raw_community, 
+              target_year_of_reference,
+              weights_method = 'area', 
+              extensive_variables = [], 
+              intensive_variables = [],
+              allocate_total = True,
+              raster = None,
+              codes = [21, 22, 23, 24],
+              force_crs_match = True):
+    """
+    Harmonize Multiples GeoData Sources with different approaches
+
+    Parameters
+    ----------
+
+    raw_community : tuple
+        Multiple GeoDataFrames given by geosnap.data.Community(..., type = "raw") (see (1) in Notes).
+    
+    target_year_of_reference : string
```

Comment:
this is a great start! It would probably also be good to allow users to define a target geom that's not necessarily present in the Community (e.g. zipcodes or the intersection of all tracts)

Reply:
Sure, I also think that the current is a little bit intuitive (also, the fact that the function doesn't harmonize the 'target_df' also, only the sources). So what do you think about refactoring the function like this:
- The user inputs a raw_community object
- In a different argument, the user inputs a set of polygons (could be any kind, for example, a specific year, polygons according to a zip code, intersection, etc.)

Then the function will harmonize ALL gdfs of the raw_community according to the generic target set of polygons that the user inputs... what do you think?

Reply:
smart

Reply:
So, re-reading your comment and mine, I think that they are pretty much the same hehehehe, ok, I think it makes sense to push forward this refactorization

Reply:
Eli, is it feasible, to assume that all datasets of the Community have a 'year' column?

Reply:
i think so, yes. The point of a Community is to provide a space-time data structure, so if you dont have more than one time period, you only need a geodataframe

with that said, though, i could imagine a temporal analysis that uses something other than census data, so your time dimension could be measured in months or quarters or something...

but for now i'd say yes; it's a safe assumption that all `Community`s will have a year col